### DISCUSSION TITLE
desktop.ini files

### DISCUSSION THREAD

**Discussion by sterisa** [abraunegg/onedrive#1745] [Q&A] [answered]
Hi,
I am getting a lot of errors like this at each sync: 
`Skipping item - invalid name (Microsoft Naming Convention): ./PicturesLibrary/Camera Roll/desktop.ini`

I am surprised I cannot find similar discussions on desktop.ini or naming convention... so I must be doing thing wrong in a new way :)

I tried adding: `skip_file =  "desktop.ini"` to the config
I tried adding: `!desktop.ini` to the sync list
But neither works
I am running ubuntu 20.04 with samba shares serving as NAS to windows machines. I have Onedrive personal, with almost 1T of data
I installed from opensuse, though I received version 2.4.14-1, instead of 2.3.13 shown as the head version for my OS

This is my config:
onedrive --display-config
```
Configuration file successfully loaded
onedrive version                       = v2.4.14-1~np1
Config path                            = /home/stefano/.config/onedrive
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /export/onedrive_sync_dir
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp|Thumbs.db|ehthumbs.db|desktop.ini|.DS_Store
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 100
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = true
sync_list contents:
# The ordering of entries is highly recommended - exclusions before inclusions
#

/Music
/PicturesLibrary
/Private
/Share
Business Shared Folders configured     = false
```

Any hint appreciated

Thanks

Stefano

**Comment by abraunegg** [comment, marked as answer]
@sterisa 
As the log out put indicates, the file is invalid due to Microsoft naming conventions.

Refer to: https://support.microsoft.com/en-us/office/restrictions-and-limitations-in-onedrive-and-sharepoint-64883a5d-228e-48f5-b3d2-eb39e07630fa#invalidfilefoldernames

**Reply by sterisa** [reply]
@abraunegg
Thanks for the prompt reply!

Related question... I tried to exclude the files from the sync (using config or sync list) but I keep getting the warnings

Stefano

**Reply by abraunegg** [reply]
@sterisa 
>I tried to exclude the files from the sync (using config or sync list) but I keep getting the warnings

This is because file validation / adherance to Microsoft requirements is done first before user exclusions.

The only way to resolve your issue is to:

* Ignore these warnings
* Fix the warnings (filename modification or delete the offending file such as 'desktop.ini')
* Use `--disable-notifications` to not use desktop notifications in monitor mode

**Reply by Interpause** [reply]
Would it be possible to selectively disable certain notifications?

**Reply by abraunegg** [reply]
@Interpause 
> Would it be possible to selectively disable certain notifications?

The only way to resolve your issue is to:

* Ignore these warnings
* Fix the warnings (filename modification or delete the offending file such as 'desktop.ini')
* Use `--disable-notifications` to not use desktop notifications in monitor mode

**Reply by Interpause** [reply]
Would you be okay with a feature request for selective notifications? When I have time, I might try to implement it myself as a patch, which could lead to a pull request.

**Comment by sterisa** [comment]
@abraunegg
Got it, thanks


On Dec 11, 2021, at 6:58 PM, abraunegg ***@***.***> wrote:

﻿
@sterisa

I tried to exclude the files from the sync (using config or sync list) but I keep getting the warnings

This is because file validation / adherance to Microsoft requirements is done first before user exclusions.

The only way to resolve your issue is to:

Ignore these warnings
Delete the offending files
